contracts/splits/src/contract.rs

```diff
@@ -1,51 +1,55 @@
 #[cfg(not(feature = "library"))]
 use cosmwasm_std::entry_point;
 use cosmwasm_std::{
-    coins, to_binary, BankMsg, Binary, Decimal, Deps, DepsMut, Env, MessageInfo, Response,
-    StdResult,
+    coins, to_binary, Addr, BankMsg, Binary, Decimal, Deps, DepsMut, Env, MessageInfo, Reply,
+    Response, StdResult, SubMsg,
 };
 use cw2::set_contract_version;
 use cw4::{Cw4Contract, Member, MemberListResponse, MemberResponse};
+use cw_utils::parse_reply_instantiate_data;
 use sg_std::NATIVE_DENOM;
 
 use crate::error::ContractError;
-use crate::msg::{ConfigResponse, ExecuteMsg, InstantiateMsg, QueryMsg};
-use crate::state::{Config, Executor, CONFIG};
+use crate::msg::{ExecuteMsg, Group, InstantiateMsg, QueryMsg};
+use crate::state::GROUP;
 
 // version info for migration info
 pub const CONTRACT_NAME: &str = "crates.io:sg-splits";
 const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
 
+const INIT_GROUP_REPLY_ID: u64 = 1;
+
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn instantiate(
     deps: DepsMut,
-    _env: Env,
+    env: Env,
     _info: MessageInfo,
     msg: InstantiateMsg,
 ) -> Result<Response, ContractError> {
-    let group_addr = Cw4Contract(deps.api.addr_validate(&msg.group_addr).map_err(|_| {
-        ContractError::InvalidGroup {
-            addr: msg.group_addr.clone(),
-        }
-    })?);
-    let weight = group_addr.total_weight(&deps.querier)?;
-    if weight == 0 {
-        return Err(ContractError::InvalidWeight { weight });
-    }
-
     set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
 
-    if let Some(Executor::Only(ref addr)) = msg.executor {
-        deps.api.addr_validate(addr.as_ref())?;
-    }
-
-    let cfg = Config {
-        group_addr,
-        executor: msg.executor,
-    };
-    CONFIG.save(deps.storage, &cfg)?;
+    let self_addr = env.contract.address;
+
+    match msg.group {
```

Comment:
I think you can put this in a function that is that is just called "group_message( )?" or something like that.

Reply:
Overkill imho.